File: scala/150.scala
import scala.collection.mutable.Stack
object Solution {
    def evalRPN(tokens: Array[String]): Int = {
        val stack = Stack[Int]()
        tokens.foreach {
            token => token match {
                case "+" => {
                    stack.push(stack.pop + stack.pop)
                }
                case "-" => {
                    val a = stack.pop()
                    val b = stack.pop()
                    stack.push(b - a)
                }
                case "*" => {
                    stack.push(stack.pop * stack.pop)
                }
                case "/" => {
                    val a = stack.pop()
                    val b = stack.pop()
                    stack.push(b / a)
                }
                case _ => {
                    stack.push(token.toInt)
                }
            }
        }
        stack.pop     
    }
}

File: leetcodeJava/150.java
package leetcodeJava;

class Solution {
    public int evalRPN(String[] tokens) {
        var stack = new LinkedList<Integer>();
        for (var token : tokens) {
            if (token.equals("-")) {
                var op1 = stack.removeLast();
                var op2 = stack.removeLast();
                stack.add(op2 - op1);
            } else if (token.equals("+")) {
                var op1 = stack.removeLast();
                var op2 = stack.removeLast();
                stack.add(op1+op2);
            } else if (token.equals("*")) {
                var op1 = stack.removeLast();
                var op2 = stack.removeLast();
                stack.add(op1*op2);
            } else if (token.equals("/")) {
                var op1 = stack.removeLast();
                var op2 = stack.removeLast();
                stack.add(op2 / op1);
            } else {
                stack.add(Integer.valueOf(token));
            }
        }
        return stack.removeLast();
    }
}
